components/dashboard/src/service/service.tsx

```diff
@@ -64,4 +65,13 @@ function getGitpodService(): GitpodService {
     return service;
 }
 
-export { getGitpodService };
+function initGitPodService(useSlowDatabase: boolean) {
+    const w = window as any;
+    const _gp = w._gp || (w._gp = {});
+    if (window.location.search.includes("service=mock")) {
+        _gp.gitpodService = require("./service-mock").gitpodServiceMock;
+    }
+    _gp.gitpodService = createGitpodService(useSlowDatabase);
+}
+
+export { getGitpodService, initGitPodService };
```

Comment:
Would you be able to explain why we're doing this?

Reply:
This function re-initializes the websocket connection - ie makes a new connection.

The `getGitpodService` function above lazily creates a ws connection, ie it will create on the first time it is called if there is no connection yet.

We want to be able to create a brand new connection on demand when the value of the `slow-database` feature flag changes.

Reply:
Thanks. Do we need the mock logic in there?

Reply:
Probably not - I took it from `getGitpodService` function above. I'm inclined to keep it.